src/main/java/org/joshsim/util/MinioHandler.java

```diff
@@ -0,0 +1,156 @@
+/**
+ * Utility class for handling Minio operations.
+ */
+
+package org.joshsim.util;
+
+import io.minio.BucketExistsArgs;
+import io.minio.MakeBucketArgs;
+import io.minio.MinioClient;
+import io.minio.UploadObjectArgs;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Handles operations related to Minio, such as bucket management and file uploads.
+ */
+public class MinioHandler {
+  private final MinioClient minioClient;
+  private final String bucketName;
+  private final String basePath;
+  private final OutputOptions output;
+
+  /**
+   * Creates a new MinioHandler.
+   *
+   * @param options The MinioOptions containing credentials and base path
+   * @param output For logging information and errors
+   * @throws Exception If Minio client creation fails
+   */
+  public MinioHandler(MinioOptions options, OutputOptions output) throws Exception {
+    if (!options.isMinioOutput()) {
+      throw new IllegalArgumentException("MinioOptions does not contain a valid Minio URL");
+    }
+
+    this.minioClient = options.getMinioClient();
+    this.bucketName = options.getBucketName();
+    this.basePath = options.getObjectPath();
+    this.output = output;
+
+    // Ensure the bucket exists
+    ensureBucketExists();
```

Comment:
I wonder if this could be behind a flag on command line. Like --ensure-bucket. Just unsure if we should do this without the user saying yes.